### DISCUSSION TITLE
WAYLAND_DISPLAY not set after wsl --update

### DISCUSSION THREAD

**Discussion by TrentonAdams** [microsoft/wslg#242] [Q&A]
The `wsl --update` seemed to work just fine, but the wayland stuff isn't working out of the gate.

```
21387.1
Windows 10 Feature Experience Pack 321.13101.0.3
```

I noticed that if I look at the login process, it shows as having variables setup for wayland, but they aren't available in the environment...
```
$ ps -awxu | grep login
root          44  0.0  0.0  10636  4704 pts/1    S    14:01   0:00 /bin/login -p -f        'DISPLAY=:0' 'HOSTTYPE=x86_64' 'PULSE_SERVER=/mnt/wslg/PulseServer' 'PWD=/home/trenta' 'TERM=xterm-256color' 'WAYLAND_DISPLAY=wayland-0' 'WSLENV=BASH_ENV/u' 'WSL_DISTRO_NAME=Ubuntu-20.04' 'WSL_INTEROP=/run/WSL/17_interop' 'XDG_RUNTIME_DIR=/mnt/wslg/runtime-dir'
$ echo $WAYLAND_DISPLAY

```

If I go and paste all those env vars in, it then works...

```
export WAYLAND_DISPLAY=wayland-0 PULSE_SERVER=/mnt/wslg/PulseServer DISPLAY=:0 XDG_RUNTIME_DIR=/mnt/wslg/runtime-dir WSL_INTEROP=/run/WSL/17_interop
gedit # this then works
```

Keep in mind I was previously using GWSL to run an X server.  I know it messes with some settings to auto set it up.

Furthermore, I cannot run mate-terminal...
```
$ mate_term=true mate-terminal
Failed to parse arguments: Cannot open display:
```

If I uninstall/re-install mate-terminal, it then comes up in the start menu, but still does not work.

I will next try backing up my current distro, and re-installing fresh to see if that works properly for things like mate-terminal.  In the mean time, if anyone has ideas to fix mate-terminal, and to get the ENV vars automatically set up, let me know.

UPDATE: 
As it turns out, mate-terminal won't do wayland, and so is not working. I'm not sure if this wayland updates comes with a working x-server, as I would think mate-terminal should work if it does. What I do know is that gedit definitely uses wayland, as setting WAYLAND_DISPLAY to something funky makes gedit not work.

What I have found is that I can continue using GWSL, and if I have WAYLAND_DISPLAY set from inside mate-terminal, gedit will indeed use wayland. I know this because I have WAYLAND_DEBUG set to 1, and it spews heavily if I have WAYLAND_DISPLAY set to wayland-0, and it does not spew anything if I don't have WAYLAND_DISPLAY set, while gedit works in both cases; clearly using the GWSL xserver or wayland as appropriate.

**Comment by TrentonAdams** [comment]
Update: I have confirmed that a brand new installation works for both mate-terminal and gedit, without the use of GWSL.  Now I have to find out why.  I suspect there's some stuff that GWSL did during install that messed things up, preventing wsl --update from working completely.

**Reply by Pololot64** [reply]
I know this is very late but note that gwsl only edits .profile and bashrc. It does not affect wsl --update
